portfolio/src/main/webapp/script.js

```diff
@@ -74,23 +74,45 @@ function hasSlides() {
     return !(slides === undefined || slides.length == 0);
 }
 
+/** Fetches commenta from the server and adds them to the DOM. */
 function getComments() {
-  fetch('/data') // sends a request to /data
-    .then(response => {
-        response.json()
-    }) // parses the response as JSON
-    .then((comments) => { // now we can reference the fields in comments
-        // Build the list of history entries.
-        const historyEl = document.getElementById("history");
-        comments.history.forEach((line) => {
-        historyEl.appendChild(createListElement(line));
-    });
+  fetch('/data')
+  .then((response) => {
+      response.json()
+    })
+  .then((comments) => {
+    const commentListElement = document.getElementById("history");
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
   });
 }
 
-/** Creates an <li> element containing text. */
-function createListElement(text) {
-  const liElement = document.createElement("li");
-  liElement.innerText = text;
-  return liElement;
-}
\ No newline at end of file
+/** Creates an element that represents a comment, including its delete button. */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const textElement = document.createElement('span');
+  textElement.innerText = comment.name +  ":\n" + comment.text;
+
+  const deleteButtonElement = document.createElement('button');
```

Comment:
minor comment: you can probably get away with naming this `deleteButton`. There are so many elements in the DOM that inclusion in the variable name does not provide very distinct signal to the reader.